javascript/react/src/components/Codebook/CodebookFieldEditor.tsx

```diff
@@ -0,0 +1,149 @@
+/*!
+ * Copyright (c) Microsoft. All rights reserved.
+ * Licensed under the MIT license. See LICENSE file in the project.
+ */
+import { DataType, VariableNature } from '@datashaper/schema'
+import { EnumDropdown } from '@essex/components'
+import { TextField } from '@fluentui/react'
+import { memo, useCallback } from 'react'
+import { If, Then } from 'react-if'
+
+import {
+	Container,
+	FieldContainer,
+	OutsideLabel,
+} from './CodebookFieldEditor.styles.js'
+import type { CodebookFieldEditorProps } from './CodebookFieldEditor.types.js'
+import {
+	CodebookFields,
+	DEFAULT_CODEBOOK_FIELDS,
+} from './CodebookFieldEditor.types.js'
+import { MappingFields } from './MappingFields.js'
+import { StatsField } from './StatsField.js'
+
+export const CodebookFieldEditor: React.FC<CodebookFieldEditorProps> = memo(
+	function CodebookFieldEditor({
+		field,
+		onChange,
+		tableView,
+		hideLabel,
+		visibleFields = DEFAULT_CODEBOOK_FIELDS,
+	}) {
+		const visibleFieldsSet = new Set<CodebookFields>()
+		visibleFields.forEach(unit => {
+			visibleFieldsSet.add(unit)
+		})
+
+		const onChangeField = useCallback(
+			(val: any) => {
+				onChange({ ...field, ...val })
+			},
+			[field, onChange],
+		)
+
+		return (
+			<Container className="codebook-column">
+				<StatsField
+					tableView={tableView}
+					onChange={onChange}
+					field={field}
+				></StatsField>
+				<If condition={visibleFieldsSet.has(CodebookFields.DisplayName)}>
+					<Then>
+						{tableView && !hideLabel && <OutsideLabel>Display</OutsideLabel>}
+						<FieldContainer className="field">
+							<TextField
+								label={!tableView && !hideLabel ? 'Display' : undefined}
+								borderless
+								disabled={field.exclude}
+								name="displayName"
+								value={field.title}
+								onChange={(_, val) => onChangeField({ title: val })}
+							/>
+						</FieldContainer>
+					</Then>
+				</If>
+				<If condition={visibleFieldsSet.has(CodebookFields.Description)}>
+					<Then>
+						{tableView && !hideLabel && (
+							<OutsideLabel>Description</OutsideLabel>
+						)}
+						<FieldContainer className="field">
+							<TextField
```

Comment:
Not sure about turning off the border on these controls. I understand the motivation, because the borders will cause it to look very cluttered. But it isn't obvious that the cell can be typed in. Maybe these need placeholder text? And maybe if it is in the entire table view we only put that placeholder text in the first column?

Reply:
See my comments elsewhere about a styles block - this could be handled by a `styles.input` that can be used to turn the borders on or off. In table mode they're off (which is set by the table editor), in field mode they're on by default.